addons/project/static/tests/project_task_state_selection_tests.js

```diff
@@ -0,0 +1,69 @@
+/** @odoo-module */
+import {
+    click,
+    getFixture,
+} from '@web/../tests/helpers/utils';
+import { setupViewRegistries } from "@web/../tests/views/helpers";
+import { start, startServer } from '@mail/../tests/helpers/test_utils';
+import { addModelNamesToFetch } from '@bus/../tests/helpers/model_definitions_helpers';
+
+addModelNamesToFetch([
+    'project.project',
+    'project.task',
+]);
+
+let target;
+
+QUnit.module('Task State Tests', {
+    beforeEach: async function () {
+        const pyEnv = await startServer();
+        const projectId = pyEnv['project.project'].create([
+            { name: "Project One" },
+        ]);
+        const userId = pyEnv['res.users'].create([
+            { name: "User One", login: 'one', password: 'one' },
+        ]);
+        pyEnv['project.task'].create([
+            { name: 'task one', project_id: projectId, state: '01_in_progress', user_ids: [userId] },
+            { name: 'task two', state: '03_approved' },
+            { name: 'task three', state: '04_waiting_normal' },
+        ]);
+        this.views = {
+            "project.task,false,kanban":
+                `<kanban js_class="project_task_kanban">
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div>
+                                <field name="state" widget="project_task_state_selection" class="project_task_state_test"/>
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+        };
+        target = getFixture();
+        setupViewRegistries();
+    }
+}, function () {
+    QUnit.test("Check whether task state widget works as intended", async function (assert) {
+        assert.expect(4);
+
+        const views = this.views;
+        const { openView } = await start({ serverData: { views } });
```

Comment:
```suggestion
        const { openView } = await start({ serverData: { this.views } });
```

Reply:
this is not allowed, check the new simplification i made :shrug: